test/distributed/fsdp/test_fsdp_checkpoint.py

```diff
@@ -201,18 +201,16 @@ def test_basic_checkpoint_end_to_end(self, cpu_offload, offload_activations):
                     else:
                         out = m(inp)
 
-                if check_offload:
-                    event_names = [event.name for event in prof.events()]
-                    offload_occured = any(
-                        offload_to_cpu_event in name for name in event_names
-                    )
-                    self.assertTrue(offload_occured)
-                loss = out.sum()
-                loss.backward()
-                losses.append(loss)
-                outputs.append(out)
-
-            self._verify_parity(losses, outputs, models)
+                    if check_offload:
```

Comment:
If I added
```
else:
    self.assertFalse(_save_on_cpu_called)
```
(and refactored the `global _save_on_cpu_called` to a higher scope (e.g. above the `for i in range(6)`) to avoid global variable definition issues), the test failed on this added assert.

Reply:
oops, yes this is because we forget to reset it for checkpoint wrapped with offloading modules. Great catch!